App Config Readme Update (#32233)
Azure/azure-sdk-for-python: sdk/appconfiguration/azure-appconfiguration-provider/README.md

In this example all configuration with empty label and the dev label are loaded. Because the dev selector is listed last, any configurations from dev take priority over those with `(No Label)` when duplicates are found.

> ## Dynamic Refresh

Reviewer: Can you add a link to the public doc so that users can get detailed information on setting this up:
https://learn.microsoft.com/en-us/azure/azure-app-configuration/enable-dynamic-configuration-python

The provider can be configured to refresh configurations from the store on a set interval. This is done by providing a `refresh_on` to the provider, which is a list of key(s) that will be watched for changes, and when they do change a refresh can happen. `refresh_interval` is the period of time in seconds between refreshes. `on_refresh_error` is a callback that will be called when a refresh fails.

```python

> from azure.appconfiguration.provider import load

Reviewer: ```suggestion
from azure.appconfiguration.provider import load, SentinelKey
```

import os

connection_string = os.environ.get("APPCONFIGURATION_CONNECTION_STRING")

def my_callback_on_fail(error):
    print("Refresh failed!")

config = load(
    connection_string=connection_string,

>     refresh_on=[SentinelKey("message")],

Reviewer: ```suggestion
    refresh_on=[SentinelKey("Sentinel")],
```

In this example all configuration with empty label and the dev label are loaded. Because the dev selector is listed last, any configurations from dev take priority over those with `(No Label)` when duplicates are found.

## Dynamic Refresh

The provider can be configured to refresh configurations from the store on a set interval. This is done by providing a `refresh_on` to the provider, which is a list of key(s) that will be watched for changes, and when they do change a refresh can happen. `refresh_interval` is the period of time in seconds between refreshes. `on_refresh_error` is a callback that will be called when a refresh fails.

```python
from azure.appconfiguration.provider import load, SentinelKey
import os

connection_string = os.environ.get("APPCONFIGURATION_CONNECTION_STRING")

def my_callback_on_fail(error):
    print("Refresh failed!")

config = load(
    connection_string=connection_string,
    refresh_on=[SentinelKey("Sentinel")],
    refresh_interval=60,
    on_refresh_error=my_callback_on_fail,
    **kwargs,
)
```

> In this example the provider will be able to be checked for changes every 60 seconds, by default 30 seconds. In order to check for changes the provider's `refresh` method needs to be called.

Reviewer: ```suggestion
In this example, the sentinel key will be checked for changes no sooner than every 60 seconds. In order to check for changes, the provider's `refresh` method needs to be called.
```
